### PR TITLE
Make error responses more flexible.

### PR DESCRIPTION
**Pull request by mduesterhoeft**
So that we could also serialize protobuf errors and a client would be able
to customize the error format.

What do you guys think about my weekend progress ;-)

### CODE REVIEW

File: router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt [hunk 1]
```diff
@@ -2,19 +2,19 @@ package io.moia.router.proto
 
 import com.google.common.net.MediaType
 import com.google.protobuf.GeneratedMessageV3
-import io.moia.router.ResponseEntity
 import io.moia.router.SerializationHandler
 import java.util.Base64
 
 class ProtoSerializationHandler : SerializationHandler {
 
     private val json = MediaType.parse("application/json")
```
Comment (blockvote) [inline router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt]: I guess we could also use `org.apache.http.entity.ContentType` here

File: router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt [hunk 2]
```diff
@@ -2,19 +2,19 @@ package io.moia.router.proto
...
-    override fun supports(acceptHeader: MediaType, response: ResponseEntity<*>): Boolean =
-        response.body is GeneratedMessageV3
+    override fun supports(acceptHeader: MediaType, body: Any): Boolean =
+        body is GeneratedMessageV3
 
-    override fun serialize(acceptHeader: MediaType, response: ResponseEntity<*>): String {
-        val message = response.body as GeneratedMessageV3
+    override fun serialize(acceptHeader: MediaType, body: Any): String {
+        val message = body as GeneratedMessageV3
+        println("$acceptHeader $message")
```
Comment (blockvote) [inline router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt]: Could this be replaced with a proper logger?

File: router/src/main/kotlin/io/moia/router/RequestHandler.kt [hunk 3]
```diff
@@ -80,89 +79,87 @@ abstract class RequestHandler : RequestHandler<APIGatewayProxyRequestEvent, APIG
...
-                .withHeaders(mapOf("Content-Type" to "application/json"))
-            else ->
-                APIGatewayProxyResponseEvent()
-                    .withBody(objectMapper.writeValueAsString(mapOf(
-                        "message" to ex.message,
-                        "code" to "INTERNAL_SERVER_ERROR"
-                    )))
-                    .withStatusCode(500)
-                    .withHeaders(mapOf("Content-Type" to "application/json"))
+                createResponse(contentType, input,
```
Comment (blockvote) [inline router/src/main/kotlin/io/moia/router/RequestHandler.kt]: I really like that it's not coupled to the `objectMapper` anymore.

File: router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt (hunk 1 above)
Comment (mduesterhoeft) [reply router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt]: I switched back to MediaType - see https://github.com/moia-dev/lambda-kotlin-request-router/pull/6 for details

File: router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt (hunk 2 above)
Comment (mduesterhoeft) [reply router-protobuf/src/main/kotlin/io/moia/router/proto/ProtoSerializationHandler.kt]: oops

**Pull request merged**